include/dice/template-library/channel.hpp

```diff
@@ -66,7 +64,7 @@ namespace dice::template_library {
                 std::lock_guard lock{queue_mutex_};
                 closed_.test_and_set(std::memory_order_release);
             }
-            queue_not_empty_.notify_one(); // notify pop() so that it does not get stuck
+            queue_not_empty_.notify_all(); // notify pop() so that it does not get stuck
```

Comment:
alternatively (or additionally), we could notify in pop on queue_not_empty_. That is already done for queue_not_full_

Reply:
Can you explain why you think this is a good idea?
Because the primary purpose of `queue_not_empty_` is to notify `pop` and the primary purpose of `queue_not_full_` is to notify `push`

Reply:
The problem with notify_one was that only one of multiple threads waiting on queue_not_empty_ (in pop()) was waken up with the notify_one before, right? So, instead the threads waiting on queue_not_empty_ (in pop()) can also wake up each other.